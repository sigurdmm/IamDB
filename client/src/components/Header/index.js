import React from 'react';
import PropTypes from 'prop-types';

const Header = ({ title, children = null }) => <header className="header">
  <h1 className="header__title">{title}</h1>
  {children}
</header>;

Header.propTypes = {
  title: PropTypes.string.isRequired,
  children: PropTypes.node,
};

export default Header;
