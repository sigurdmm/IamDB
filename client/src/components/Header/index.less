@import '../../constants';

.header {
  box-sizing: border-box;
  display: flex;

  justify-content: space-between;

  &__title {
    text-decoration: none;

    h1 {
      .text-shadow-1;
    }
  }
}
