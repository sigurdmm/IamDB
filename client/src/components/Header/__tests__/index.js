import { render } from 'enzyme';

import Header from '../index';

describe('<Header/>', () => {
  it('renders without children', () => {
    const wrapper = render(<Header title="Hello world"/>);

    expect(wrapper).toMatchSnapshot();
  });

  it('renders with children', () => {
    const wrapper = render(<Header title="Hello world">
      <nav>
        <a href="#1">Link 1</a>
        <a href="#2">Link 2</a>
      </nav>
    </Header>);

    expect(wrapper).toMatchSnapshot();
  });
});
