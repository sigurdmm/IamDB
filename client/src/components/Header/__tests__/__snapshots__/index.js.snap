// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header/> renders with children 1`] = `
<header
  class="header"
>
  <h1
    class="header__title"
  >
    Hello world
  </h1>
  <nav>
    <a
      href="#1"
    >
      Link 1
    </a>
    <a
      href="#2"
    >
      Link 2
    </a>
  </nav>
</header>
`;

exports[`<Header/> renders without children 1`] = `
<header
  class="header"
>
  <h1
    class="header__title"
  >
    Hello world
  </h1>
</header>
`;
